src/main/java/tr/com/srdc/cda2fhir/transform/util/impl/LocalBundleInfo.java

```diff
@@ -5,19 +5,29 @@
 
 import org.hl7.fhir.dstu3.model.Identifier;
 import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.openhealthtools.mdht.uml.hl7.datatypes.CD;
 import org.openhealthtools.mdht.uml.hl7.datatypes.II;
 
 import tr.com.srdc.cda2fhir.transform.IResourceTransformer;
+import tr.com.srdc.cda2fhir.transform.entry.CDAIIResourceMaps;
 import tr.com.srdc.cda2fhir.transform.entry.IEntityInfo;
+import tr.com.srdc.cda2fhir.transform.entry.IEntityResult;
 import tr.com.srdc.cda2fhir.transform.util.IBundleInfo;
-import tr.com.srdc.cda2fhir.transform.util.ICDAIIMapSource;
+import tr.com.srdc.cda2fhir.transform.util.IResult;
 
 public class LocalBundleInfo implements IBundleInfo {
 	private IBundleInfo bundleInfo;
 	private CDAIIMap<IEntityInfo> entities = new CDAIIMap<IEntityInfo>();
+	private CDAIIResourceMaps<IBaseResource> resourceMaps = new CDAIIResourceMaps<IBaseResource>();
+	private CDACDMap<IBaseResource> cdMap = new CDACDMap<IBaseResource>();
 
 	public LocalBundleInfo(IBundleInfo bundleInfo) {
 		this.bundleInfo = bundleInfo;
+		entities.put(bundleInfo.getEntities());
+		resourceMaps.put(bundleInfo.getResourceMaps());
+		cdMap.put(bundleInfo.getCDMap());
+
```

Comment:
When new medications are found inside of manufacturedProduct2Medication, they aren't added to the bundle info, they're added to the EntryResult. 

In the tMedicationActivity2MedicationStatement there are two calls to the manufacturedProduct2Medication function. First for the consumable, and then again inside of medicationSupplyOrder2MedicationRequest. 

We need to update whatever bundle info is used for the consumable before it's used for the medicationSupplyOrder. I used the localBundleInfo inside of MedActivity2MedicationStatement because I didn't want to update bundleInfo (the parameter) before using it again in the call to medicationSupplyOrder2MedicationRequest. Instead, all updated II mappings and CD mappings would be collected in the local bundle info and then transferred to the MedicationStatment EntryResult to be returned.

Kind of hard to describe, but I couldn't figure out a way to do it without a local bundle info to update the mappings